boards/pic32-wifire/pic32_config_settings.c

```diff
@@ -0,0 +1,419 @@
+    & (~_DEVCFG0_POSCBOOST_MASK   |  0x1      << _DEVCFG0_POSCBOOST_POSITION)
+    & (~_DEVCFG0_EJTAGBEN_MASK    |  0x1      << _DEVCFG0_EJTAGBEN_POSITION);
+
+/*
+ * uint32_t DEVCFG0_B2  __attribute__ ((section(".devcfg0_b2")))
+ * = 0xFFFFF7D7;
+ * uint32_t ADEVCFG0_B2  __attribute__ ((section(".adevcfg0_b2")))
+ * = 0xFFFFF7D7;
+ *
+ */
+
+
+/*
+ * DEVCP0_LA  @ 0x1FC0FFDC (lower alias)
+ * ADEVCP0_LA @ 0x1FC0FF5C (alternate devcp0 in lower alias)
+ * DEVCP0_B1  @ 0x1FC4FFDC (config space 1)
+ * ADEVCP0_B1 @ 0x1FC4FF5C (alternate devcp0 in config space 1)
+ * DEVCP0_B2  @ 0x1FC6FFDC (config space 1)
+ * ADEVCP0_B2 @ 0x1FC6FF5C (alternate devcp0 in config space 2
+ *
+ * CP   OFF Code Protect    Protection Disabled, unused bits must be 1.
+ */
+
+volatile uint32_t DEVCP0_LA  __attribute__ ((used, section(".devcp0_la")))  =
+    0xFFFFFFFF | _DEVCP0_CP_MASK;
+volatile uint32_t ADEVCP0_LA __attribute__ ((used, section(".adevcp0_la"))) =
+    0xFFFFFFFF | _DEVCP0_CP_MASK;
+volatile uint32_t DEVCP0_B1  __attribute__ ((used, section(".devcp0_b1")))  =
+    0xFFFFFFFF | _DEVCP0_CP_MASK;
+volatile uint32_t ADEVCP0_B1 __attribute__ ((used, section(".adevcp0_b1"))) =
+    0xFFFFFFFF | _DEVCP0_CP_MASK;
+/* not needed by default */
+/* uint32_t DEVCP0_B2   __attribute__ ((section(".devcp0_b1"))) = 0xFFFFFFFF; */
+/* uint32_t ADEVCP0_B2  __attribute__ ((section(".adevcp0_b1"))) = 0xFFFFFFFF; */
+
+/*
+ * SEQ_B1[0..3] @ 1FC0FFF0
+ * SEQ_B1[0..3] @ 1FC4FFF0
+ *
+ * TSEQ     Boot Flash True Sequence Number
+ * CSEQ     Boot Flash Complement Sequence Number
+ */
+
+volatile uint32_t SEQ_LA[4] __attribute__ ((used, section(".seq_la"))) =
+    { 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF };
+volatile uint32_t SEQ_B1[4] __attribute__ ((used, section(".seq_b1"))) =
+    { 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF };
+/*
+ * Not needed by default:
+ * uint32_t SEQ_B2[4] __attribute__ ((section(".seq_b2"))) =
+ *   {0xFFFFFFFF,0xFFFFFFFF,0xFFFFFFFF,0xFFFFFFFF};
+ */
+
+
+/*
+ * STUPIDLY Microchip has hard coded the MSB bit of devsign to 0, So even if
+ * you erase the whole device, everything returns 0xFFFFFFF except this 1
+ * register (and its alternate) which return 0x7FFFFFF!!
+ *
+ * We set it in the output image so verification doesn't fail
+ *
+ * DEVSIGN0 @ 0xBFC0FFEC
+ * ADEVSIGN0 @ 0xBFC0FF6C
+ *
+ */
+
+volatile uint32_t DEVSIGN_LA  __attribute__ ((used, section(".devsign_la")))  = 0x7FFFFFFF;
+volatile uint32_t ADEVSIGN_LA __attribute__ ((used, section(".adevsign_la"))) = 0x7FFFFFFF;
+volatile uint32_t DEVSIGN_B1  __attribute__ ((used, section(".devsign_b1")))  = 0x7FFFFFFF;
+volatile uint32_t ADEVSIGN_B1 __attribute__ ((used, section(".adevsign_b1"))) = 0x7FFFFFFF;
+volatile uint32_t DEVSIGN_B2  __attribute__ ((used, section(".devsign_b2")))  = 0x7FFFFFFF;
+volatile uint32_t ADEVSIGN_B2 __attribute__ ((used, section(".adevsign_b2"))) = 0x7FFFFFFF;
+
+
+/*
+ * Without a reference to this function from elsewhere LD throws the whole
```

Comment:
Probably has to do with missing '--whole-archive'. Could you please check if #5757 removes the need for this?